src/magma/vk/pipeline/PipelineInfo.cpp

```diff
@@ -0,0 +1,151 @@
+#include "magma/vk/pipeline/PipelineInfo.h"
+
+#include "magma/vk/vulkan_common.h"
+#include <vulkan/vulkan.hpp>
+
+PipelineLayoutInfo::PipelineLayoutInfo()
+        : _pipelineLayoutInfo()
+{}
+
+PipelineLayoutInfo::PipelineLayoutInfo(const std::vector<vk::DescriptorSetLayout> &descriptorSetLayouts)
+        : _pipelineLayoutInfo({}, descriptorSetLayouts)
+{}
+
+PipelineLayoutInfo::PipelineLayoutInfo(const vk::DescriptorSetLayout &descriptorSetLayout)
+        : _pipelineLayoutInfo({}, 1, &descriptorSetLayout)
+{}
+
+PipelineInfo::PipelineInfo(vk::Extent2D extent) 
+        : _resolution(extent) {
+
+    //vertex input info
+    //use setVertexInputInfo to change
+    _vertexInput.vertexBindingDescriptionCount = 0;
+    _vertexInput.pVertexBindingDescriptions = nullptr; // Optional
+    _vertexInput.vertexAttributeDescriptionCount = 0;
+    _vertexInput.pVertexAttributeDescriptions = nullptr; // Optional
+
+    //topology
+    _inputAssembly.topology = vk::PrimitiveTopology::eTriangleList;
+    _inputAssembly.primitiveRestartEnable = false;
+
+    //viewport
+    _viewport.x = 0.0f;
+    _viewport.y = 0.0f;
+    _viewport.width  = (float) extent.width;
+    _viewport.height = (float) extent.height;
+    _viewport.minDepth = 0.0f;
+    _viewport.maxDepth = 1.0f;
+
+    //scissor
+    _scissor.offset = vk::Offset2D(0, 0);
+    _scissor.extent = extent;
+
+    _viewportState.viewportCount = 1;
+    _viewportState.pViewports = &_viewport;
+    _viewportState.scissorCount = 1;
+    _viewportState.pScissors = &_scissor;
+
+    //rasterizer
+    _rasterization.depthClampEnable = false; //to set true requires gpu feature
```

Comment:
_rasterizer was shorter and it's a noun which is better for a member name than a verb